unstructured/ingest/v2/processes/connectors/milvus.py

```diff
@@ -0,0 +1,200 @@
+        defaults = {"is_continuation": False}
+
+        if metadata := data.pop("metadata", None):
+            data.update(flatten_dict(metadata, keys_to_omit=["data_source_record_locator"]))
+        for datetime_column in datetime_columns:
+            if datetime_column in data:
+                data[datetime_column] = cls.parse_date_string(data[datetime_column])
+        for json_dumps_field in json_dumps_fields:
+            if json_dumps_field in data:
+                data[json_dumps_field] = json.dumps(data[json_dumps_field])
+        for default in defaults:
+            if default not in data:
+                data[default] = defaults[default]
+
+    def run(
+        self,
+        elements_filepath: Path,
+        file_data: FileData,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        with open(elements_filepath) as elements_file:
+            elements_contents: list[dict[str, Any]] = json.load(elements_file)
+        for element in elements_contents:
+            self.conform_dict(data=element)
+
+        output_path = Path(output_dir) / Path(f"{output_filename}.json")
+        output_path.parent.mkdir(parents=True, exist_ok=True)
+        with output_path.open("w") as output_file:
+            json.dump(elements_contents, output_file, indent=2)
+        return output_path
+
+
+@dataclass
+class MilvusUploaderConfig(UploaderConfig):
+    collection_name: str
+    num_of_processes: int = 4
+
+
+@dataclass
+class MilvusUploader(Uploader):
+    connection_config: MilvusConnectionConfig
+    upload_config: MilvusUploaderConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    def upload(self, content: UploadContent) -> None:
+        file_extension = content.path.suffix
+        if file_extension == ".json":
+            self.upload_json(content=content)
+        elif file_extension == ".csv":
+            self.upload_csv(content=content)
+        else:
+            raise ValueError(f"Unsupported file extension: {file_extension}")
+
+    @requires_dependencies(["pymilvus"], extras="milvus")
+    def insert_results(self, data: Union[dict, list[dict]]):
+        from pymilvus import MilvusException
+
+        logger.debug(
+            f"uploading {len(data)} entries to {self.connection_config.db_name} "
+            f"db in collection {self.upload_config.collection_name}"
+        )
+        client = self.connection_config.get_client()
+
+        try:
+            res = client.insert(collection_name=self.upload_config.collection_name, data=data)
+        except MilvusException as milvus_exception:
+            raise WriteError("failed to upload records to milvus") from milvus_exception
+        if "err_count" in res and isinstance(res["err_count"], int) and res["err_count"] > 0:
+            err_count = res["err_count"]
+            raise WriteError(f"failed to upload {err_count} docs")
+
+    def upload_csv(self, content: UploadContent) -> None:
```

Comment:
This seems strange. I forced number_of_processes to 1. Which ran this line.

And then it tried to read_csv but was a .json file.

And I got this:
*** pandas.errors.ParserError: Error tokenizing data. C error: Expected 1 fields in line 3, saw 2